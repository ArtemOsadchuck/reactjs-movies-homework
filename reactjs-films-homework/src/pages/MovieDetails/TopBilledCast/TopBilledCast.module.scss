.actorWrapper {
  color: rgb(247, 247, 247);
  display: flex;
  flex-direction: column;
  height: 13rem;
  justify-content: space-between;
  width: 7rem;

  .actorImg {
    border-radius: 5%;
  }

  .actorName {
    font-size: 0.8rem;
    text-align: left;
  }

  .actorCharacter {
    font-size: 0.7rem;
    letter-spacing: 0.5px;
    text-align: left;
  }
}
