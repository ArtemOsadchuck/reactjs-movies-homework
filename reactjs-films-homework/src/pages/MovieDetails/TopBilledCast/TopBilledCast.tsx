import React from 'react';
import styles from './TopBilledCast.module.scss';

export interface ITopBilledCastProp {
  id: number;
  name: string;
  profile_path: string;
  character: string;
  popularity: number;
}
export interface ITopBilledCast {
  props: ITopBilledCastProp;
}

const TopBilledCast: React.FC<ITopBilledCast> = ({ props }) => {
  const { id, name, profile_path, character } = props;
  const urlImg = `https://image.tmdb.org/t/p/w200/${profile_path}`;

  return (
    <a className={styles.actorWrapper} id={`${id}`} href="/">
      <img className={styles.actorImg} height="160px" src={urlImg} alt={name} />
      <p className={styles.actorName}>
        <strong>{name}</strong>
      </p>
      <p className={styles.actorCharacter}>{character}</p>
    </a>
  );
};

export default TopBilledCast;
