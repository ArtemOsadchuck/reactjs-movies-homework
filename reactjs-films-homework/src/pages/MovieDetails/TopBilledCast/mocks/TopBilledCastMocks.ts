const TopBilledCastMocks = {
  adult: false,
  gender: 2,
  id: 192,
  known_for_department: 'Acting',
  name: 'Morgan Freeman',
  original_name: 'Morgan Freeman',
  popularity: 17.844,
  profile_path: '/oIciQWr8VwKoR8TmAw1owaiZFyb.jpg',
  cast_id: 4,
  character: "Ellis Boyd 'Red' Redding",
  credit_id: '52fe4231c3a36847f800b135',
  order: 1,
};

export default TopBilledCastMocks;
