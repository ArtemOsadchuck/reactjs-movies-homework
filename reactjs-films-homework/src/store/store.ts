import rootStore from './rootStore/rootStore';

export type RootState = ReturnType<typeof rootStore.getState>;
export type AppDispatch = typeof rootStore.dispatch;
export default rootStore;
