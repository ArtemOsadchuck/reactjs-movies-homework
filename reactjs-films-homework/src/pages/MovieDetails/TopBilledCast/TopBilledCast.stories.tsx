import React from 'react';
import { Story } from '@storybook/react';

import TopBilledCast from './TopBilledCast';
import TopBilledCastMocks from './mocks';

import RouterWrapper from '../../../__testsUtils__/routerHoc';
import StoreWrapper from '../../../__testsUtils__/storeHoc';

export default {
  title: 'TopBilledCast',
  component: TopBilledCast,
};

const TemplateUrl: Story<typeof TopBilledCast> = () => (
  <StoreWrapper>
    <RouterWrapper url="/">
      <TopBilledCast props={TopBilledCastMocks} />
    </RouterWrapper>
  </StoreWrapper>
);

export const TopBilledCastStory = TemplateUrl.bind({});
