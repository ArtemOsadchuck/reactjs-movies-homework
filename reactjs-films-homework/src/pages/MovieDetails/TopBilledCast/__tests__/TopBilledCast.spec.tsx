import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';

import TopBilledCast from '../TopBilledCast';
import TopBilledCastMocks from '../mocks';

import { Provider } from 'react-redux';
import store from '../../../../store/store';

describe('TopBilledCast', () => {
  let fragment: any;

  const history = createMemoryHistory();
  history.push('/');

  beforeEach(async () => {
    const { asFragment } = await render(
      <Provider store={store}>
        <Router history={history}>
          <TopBilledCast props={TopBilledCastMocks} />
        </Router>
      </Provider>
    );
    fragment = asFragment();
  });

  afterEach(() => cleanup());

  it('TopBilledCast snapshot', () => {
    expect(fragment).toMatchSnapshot();
  });

  it('TopBilledCast img must have attribute:', () => {
    expect(screen.getByRole(/img/i)).toHaveAttribute('alt');
    expect(screen.getByRole(/img/i)).toHaveAttribute('height');
    expect(screen.getByRole(/img/i)).toHaveAttribute('src');
  });
});
