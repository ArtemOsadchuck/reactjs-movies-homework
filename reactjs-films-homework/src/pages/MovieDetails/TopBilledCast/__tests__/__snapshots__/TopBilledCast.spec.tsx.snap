// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBilledCast TopBilledCast snapshot 1`] = `
<DocumentFragment>
  <a
    class="actorWrapper"
    href="/actor-profile/?actor-id=192"
    id="192"
  >
    <img
      alt="Morgan Freeman"
      class="actorImg"
      height="160px"
      src="https://image.tmdb.org/t/p/w500//oIciQWr8VwKoR8TmAw1owaiZFyb.jpg"
    />
    <p
      class="actorName"
    >
      <strong>
        Morgan Freeman
      </strong>
    </p>
    <p
      class="actorCharacter"
    >
      Ellis Boyd 'Red' Redding
    </p>
  </a>
</DocumentFragment>
`;
